resources/validations/bin/validate_with_schematron.sh

```diff
@@ -1,69 +1,126 @@
 #!/usr/bin/env bash
-
 set -o pipefail
 
-if [ ! -e  "$1" ]; then
+# output root defaults to report folder relative to this script
+OUTPUT_ROOT="report/schematron"
+# schematron directory validate the file with each .sch found defaults to src/*.sch relative to this script
+SCHEMA_LOCATION_DIR="src"
+##
+## options ###################################################################
+##
+while echo "$1" | grep -- ^- > /dev/null 2>&1; do
+    case "$1" in
+        # input file to validate
+        -f)
+            shift
+            DOC_TO_VALIDATE="$1"
+            ;;
+        # saxon version
+        -v)
+            if test -n "$SAXON_CP"; then
+                echo "SAXON_CP is set to ${SAXON_CP} as an environment variable setting version using -v is invalid"
+                exit 1
+            else
+                shift
+                SAXON_VERSION="$1"
+            fi
+            ;;
+        # schema directory location
+        -s)
+            shift
+            SCHEMA_LOCATION_DIR="$1"
+            ;;
+        # output directory root
+        -o)
+            shift
+            OUTPUT_ROOT="$1"
+            ;;
+        # Unknown option!
+        -*)
+            usage "Error: Unknown option: $1"
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+echo output dir "${OUTPUT_ROOT}"
+echo to val "$DOC_TO_VALIDATE";
+if test ! -e  "$DOC_TO_VALIDATE" ; then
     echo "no file input for report, exiting"
     exit 1
+else 
+    echo "doc requested to be validated: ${DOC_TO_VALIDATE}"
 fi
-DOC_TO_VALIDATE="$1"
-echo "doc requested to be validated: ${DOC_TO_VALIDATE}"
 
 # Delete pre-existing XSLT report
 rm -rf target/*.xsl;
 
-SAXON_VERSION=$2
+#if version not specified default
 SAXON_VERSION=${SAXON_VERSION:-10.2}
 SAXON_OPTS="${SAXON_OPTS:-allow-foreign=true}"
 
 echo "using saxon version ${SAXON_VERSION}"
 
-mvn -q org.apache.maven.plugins:maven-dependency-plugin:2.1:get \
-    -DrepoUrl=https://mvnrepository.com/ \
-    -DartifactId=Saxon-HE \
-    -DgroupId=net.sf.saxon \
-    -Dversion="${SAXON_VERSION}"
+saxonLocation=saxon/Saxon-HE/"${SAXON_VERSION}"/Saxon-HE-"${SAXON_VERSION}".jar
+if test -n "$SAXON_CP" ; then
+    echo SAXON_CP env variable used is "${SAXON_CP}"
+elif command -v mvn &> /dev/null ;then
+    mvn -q org.apache.maven.plugins:maven-dependency-plugin:2.1:get \
+        -DrepoUrl=https://mvnrepository.com/ \
+        -DartifactId=Saxon-HE \
+        -DgroupId=net.sf.saxon \
+        -Dversion="${SAXON_VERSION}"
+    SAXON_CP=~/.m2/repository/net/sf/${saxonLocation}
+elif command -v curl &> /dev/null; then
+    SAXON_CP=lib/Saxon-HE-"${SAXON_VERSION}".jar
```

Comment:
We should add to `.gitignore` if we put it there potentially, but I like this approach.

Reply:
it's part of the PR already